src/main/java/com/atlassian/jira/cloud/jenkins/deploymentinfo/service/FreestyleChangeLogIssueKeyExtractor.java

```diff
@@ -15,24 +17,47 @@
 
 public class FreestyleChangeLogIssueKeyExtractor implements FreestyleIssueKeyExtractor {
     @Override
-    public Set<String> extractIssueKeys(final AbstractBuild<?, ?> freestyleBuild) {
+    public Set<String> extractIssueKeys(
+            final AbstractBuild<?, ?> freestyleBuild, final PipelineLogger pipelineLogger) {
 
         final Set<IssueKey> allIssueKeys = new HashSet<>();
-        final List<ChangeLogSet<? extends ChangeLogSet.Entry>> changeSets = freestyleBuild.getChangeSets();
+        final List<ChangeLogSet<? extends ChangeLogSet.Entry>> changeSets =
+                freestyleBuild.getChangeSets();
         for (ChangeLogSet<? extends ChangeLogSet.Entry> changeSet : changeSets) {
             final Object[] changeSetEntries = changeSet.getItems();
             for (Object item : changeSetEntries) {
                 final ChangeLogSet.Entry changeSetEntry = (ChangeLogSet.Entry) item;
                 if (changeSetEntry instanceof GitChangeSet) {
-                    allIssueKeys.addAll(
-                            IssueKeyStringExtractor.extractIssueKeys(((GitChangeSet) changeSetEntry).getComment()));
+                    String comment = ((GitChangeSet) changeSetEntry).getComment();
+                    Set<IssueKey> issueKeys = IssueKeyStringExtractor.extractIssueKeys(comment);
+                    allIssueKeys.addAll(issueKeys);
+                    pipelineLogger.debug(
+                            String.format(
+                                    "Extracted issue keys from GitChangeSet comment '%s': %s",
+                                    comment, Arrays.toString(issueKeys.toArray())));
                 }
-                allIssueKeys.addAll(IssueKeyStringExtractor.extractIssueKeys(changeSetEntry.getMsg()));
+
+                String message = changeSetEntry.getMsg();
+                Set<IssueKey> issueKeys = IssueKeyStringExtractor.extractIssueKeys(message);
+                allIssueKeys.addAll(issueKeys);
+                pipelineLogger.debug(
+                        String.format(
+                                "Extracted issue keys from ChangeLogSet message '%s': %s",
+                                message, Arrays.toString(issueKeys.toArray())));
+
                 if (allIssueKeys.size() >= ISSUE_KEY_MAX_LIMIT) {
+                    pipelineLogger.warn(
+                            String.format(
+                                    "Not extracting any more issues as the maximum of %d has been reached!",
+                                    ISSUE_KEY_MAX_LIMIT));
                     break;
```

Comment:
In this file, I just added some log events around the issue key extraction, so that we can see from the logs if and which issues were extracted.